packages/api-client-core/src/GadgetRecord.ts

```diff
@@ -9,66 +9,75 @@ export enum ChangeTracking {
 
 export type RecordShape = Record<string, any> | null | undefined | void;
 
+const kFields = Symbol.for("g/fields");
+const kInstantiatedFields = Symbol.for("g/if");
+const kPersistedFields = Symbol.for("g/pf");
+const kFieldKeys = Symbol.for("g/fk");
+const kTouched = Symbol.for("g/t");
+
 /** Represents one record returned from a high level Gadget API call */
-export class GadgetRecordImplementation<Shape extends RecordShape> {
-  private __gadget = {
-    fields: {} as any,
-    instantiatedFields: {} as any,
-    persistedFields: {} as any,
-    fieldKeys: [] as string[],
-    fieldKeysTracker: {} as Record<string, boolean>,
-    touched: false,
-  };
+export class GadgetRecord_<Shape extends RecordShape> {
+  /** Storage of the actual keys and values of this record */
+  [kFields]: Record<string, any> = {};
+  /** Storage of the keys and values of this record at the time it was instantiated */
+  [kInstantiatedFields]: Record<string, any> = {};
+  /** Storage of the keys and values of this record at the time it was last persisted */
+  [kPersistedFields]: Record<string, any> = {};
+  /** Storage of the keys and values of this record at the time it was last persisted */
+  [kFieldKeys]: Set<string>;
+  [kTouched] = false;
 
   private empty = false;
 
   constructor(data: Shape) {
-    this.__gadget.instantiatedFields = cloneDeep(data);
-    this.__gadget.persistedFields = cloneDeep(data);
-    Object.assign(this.__gadget.fields, data);
+    this[kInstantiatedFields] = cloneDeep(data) ?? {};
+    this[kPersistedFields] = cloneDeep(data) ?? {};
```

Comment:
I tried making this lazy but it breaks our contracts around detecting nested changes, so we still gotta do this eagerly on instantiation :(

Reply:
Yup, or just only doing the copy the first time we need to write something. But, we have tests that say "modifying an array property marks that property as changed", which means we need to either proxy nested stuff in order to catch the write to the nested doodad, or just keep eagerly deepcloning. It... sucks.